Getting runtime error while want to see help for 'date'

# Description
I want to see the description of the date generator and it gives me a runtime error!

# Steps To Reproduce
Use Version `v1.1.0` of fakedata installed with `brew install lucapette/tap/fakedata` on macOS 10.12.6.
Type `fakedata -g date` into a Terminal.

# Expected result
The description of the date generator

# Actual result
The following was printed in the Terminal:
```
Description: random date in the format YYYY-MM-DD. By default, it generates dates in the last year

Example:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x112b8ad]

goroutine 1 [running]:
main.main()
	/Users/lucapette/src/github.com/lucapette/fakedata/main.go:127 +0x9dd
```

@lucapette I took a quick look and it seems to be related to generators which take arguments. 

```sh
$ fakedata -g int
Description: positive integer between 1 and 1000

Example:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x112b8ad]

goroutine 1 [running]:
main.main()
	/Users/lucapette/src/github.com/lucapette/fakedata/main.go:127 +0x9dd

```

```sh
$ fakedata -g enum
Description: value from an enum. By default, the enum is foo,bar,baz. It accepts a list of comma-separated values

Example:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x112b8ad]

goroutine 1 [running]:
main.main()
	/Users/lucapette/src/github.com/lucapette/fakedata/main.go:127 +0x9dd
```

--- a/integration/cli_test.go
+++ b/integration/cli_test.go
@@ -7,6 +7,7 @@ import (
 	"os"
 	"os/exec"
 	"path/filepath"
+	"regexp"
 	"runtime"
 	"testing"
 
@@ -153,7 +154,7 @@ func TestCLI(t *testing.T) {
 			"unknown format",
 			[]string{"-f=sqll", "-t=USERS", "int:42,42", "enum:foo,foo"},
 			"unknown-format.golden",
-			false,
+			true,
 		},
 	}
 
@@ -173,8 +174,38 @@ func TestCLI(t *testing.T) {
 			}
 			expected := golden.load()
 
-			if !reflect.DeepEqual(actual, expected) {
+			if !reflect.DeepEqual(expected, actual) {
 				t.Fatalf("diff: %v", diff(expected, actual))
+			}
+		})
+	}
+}
+
+func TestGeneratorDescription(t *testing.T) {
+	tests := []struct {
+		name string
+		args []string
+	}{
+		{"simple generator", []string{"-g", "name.first"}},
+		{"custom generator", []string{"-g", "int"}},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			cmd := exec.Command(binaryPath, tt.args...)
+			output, err := cmd.CombinedOutput()
+			if err != nil {
+				t.Fatalf("test run returned an error: %v\n%s", err, output)
+			}
+
+			actual := string(output)
+			matched, err := regexp.MatchString("Description:", actual)
+			if err != nil {
+				t.Fatalf("could not match actual: %v", err)
+			}
+
+			if !matched {
+				t.Fatalf("expected %s to match description, but did not", actual)
 			}
 		})
 	}

--- a/main.go
+++ b/main.go
@@ -124,7 +124,17 @@ func main() {
 		if generator := generators.FindByName(*generatorFlag); generator != nil {
 			fmt.Printf("Description: %s\n\nExample:\n\n", generator.Desc)
 			for i := 0; i < 5; i++ {
-				fmt.Println(generator.Func())
+				fn := generator.Func
+				if generator.IsCustom() {
+					custom, err := generator.CustomFunc("")
+					if err != nil {
+						fmt.Printf("could not generate example: %v", err)
+						os.Exit(1)
+					}
+
+					fn = custom
+				}
+				fmt.Println(fn())
 			}
 		}
 		os.Exit(0)
@@ -162,7 +172,7 @@ func main() {
 	if err != nil {
 		fmt.Printf("%v\n\n", err)
 		flag.Usage()
-		os.Exit(0)
+		os.Exit(1)
 	}
 
 	for i := 0; i < *limitFlag; i++ {

